src/ocrd_anybaseocr/cli/ocrd_anybaseocr_layout_analysis.py

```diff
@@ -0,0 +1,227 @@
+from functools import cached_property
+import sys
+import pickle
+from typing import Optional
+import numpy as np 
+import warnings
+
+from ocrd_models import OcrdFileType
+warnings.filterwarnings('ignore',category=FutureWarning) 
+from collections import defaultdict
+
+import click
+from ocrd.decorators import ocrd_cli_options, ocrd_cli_wrap_processor
+
+from ocrd import OcrdPage, OcrdPageResult, Processor
+from ocrd_modelfactory import page_from_file
+from ocrd_utils import getLogger, resource_filename
+
+from pathlib import Path
+from PIL import Image
+
+from lxml import etree as ET
+
+from ocrd_models.constants import (
+    NAMESPACES as NS,
+    TAG_METS_DIV,
+    TAG_METS_STRUCTMAP,
+)
+
+from ..tensorflow_importer import tf
+from ..utils import pil2array
+from tensorflow.keras.models import load_model
+
+TAG_METS_STRUCTLINK = '{%s}structLink' % NS['mets']
+TAG_METS_SMLINK = '{%s}smLink' % NS['mets']
+
+class OcrdAnybaseocrLayoutAnalyser(Processor):
+    
+    @cached_property
+    def executable(self):
+        return 'ocrd-anybaseocr-layout-analysis'
+
+    def setup(self):
+        if not tf.config.list_physical_devices('GPU'):
+            self.logger.error("Your system has no CUDA installed. No GPU detected.")
+
+        self.last_result = [] 
+        self.logID = 0 # counter for new key
+        self.logIDs = defaultdict(int) # dict to keep track of previous keys for labels other then chapter or section
+        self.log_id = 0 # var to keep the current ongoing key
+        self.log_links = {}
+        self.first = None
+        
+        assert self.parameter
+        if 'model_path' not in self.parameter:
+            self.parameter = {'model_path': str(resource_filename(f'ocrd_anybaseocr.models', 'structure_analysis')), **self.parameter}
+        if 'class_mapping_path' not in self.parameter:
+            self.parameter = {'class_mapping_path': str(resource_filename(f'ocrd_anybaseocr.models', 'mapping_densenet.pickle')), **self.parameter}
+
+        model_path = Path(self.resolve_resource(self.parameter['model_path']))
+        class_mapper_path = Path(self.resolve_resource(self.parameter['class_mapping_path']))
+        if not model_path.exists():
+            self.logger.critical("Layout classfication `model_path` was not found at '%s'", model_path)
+            sys.exit(1)
+        self.logger.info('Loading model from file %s', str(model_path))
+        self.model = self.create_model(str(model_path))
+        # load the mapping
+        pickle_in = open(str(class_mapper_path), "rb")
+        class_indices = pickle.load(pickle_in)
+        self.label_mapping = dict((v,k) for k,v in class_indices.items())
+
+    def process_page_file(self, *input_files : Optional[OcrdFileType]) -> None:
+        self.logger.info("Overridden process_page_file")
+        for input_file in input_files:
```

Comment:
Instead of this iteration pattern, we should assert our way out of multi-fileGrp input here.